jujugui/static/gui/src/app/components/generic-input/test-generic-input.js

```diff
@@ -392,4 +424,26 @@ describe('GenericInput', function() {
     output.props.children[1].props.onBlur();
     assert.equal(updateModelName.callCount, 1);
   });
+
+  it('can call the passed keyup function', () => {
+    const updateModelName = sinon.stub();
+    const renderer = jsTestUtils.shallowRender(
+      <juju.components.GenericInput
+        disabled={false}
+        label="Region"
+        placeholder="us-central-1"
+        required={true}
+        onKeyUp={updateModelName}
+        ref="templateRegion"
+        validate={[{
+          regex: /\S+/,
+          error: 'This field is required.'
+        }]}
+        value="default" />, true);
+    const instance = renderer.getMountedInstance();
+    instance.refs = {field: {value: ''}};
+    let output = renderer.getRenderOutput();
+    output.props.children[1].props.onKeyUp();
```

Comment:
If we cannot simulate an actual keyup then this test is asserting that the provided callback is called when called on the output props. Maybe we should rename the test.